.circleci/config.yml

```diff
@@ -292,15 +302,17 @@ jobs:
           destination: node3/
 
   docker_image_build:
-    <<: *container_config
+    <<: *machine_config
     steps:
       - checkout
-      - *setup_remote_docker
-      - *install_docker_client
       - run:
-          name: Build Docker image
+          name: Build image and smoke test the nodes
           command: |
-            ~/bin/docker build .
+            docker-compose build node1
+            docker-compose up -d
+            sleep 10
```

Comment:
I am a bit concerned this may cause failures in following line if too short. I suggest improving with a wait/until-like construct - though not a blocker for this PR (can be tracked and addressed later as needed).